Source/WebKit/UIProcess/Extensions/WebExtensionDynamicScripts.h

```diff
@@ -66,6 +75,46 @@ class InjectionResultHolder : public RefCounted<InjectionResultHolder> {
     InjectionResults results;
 };
 
+class WebExtensionRegisteredScript : public RefCounted<WebExtensionRegisteredScript> {
+    WTF_MAKE_NONCOPYABLE(WebExtensionRegisteredScript);
+    WTF_MAKE_FAST_ALLOCATED;
+
+public:
+    template<typename... Args>
+    static Ref<WebExtensionRegisteredScript> create(Args&&... args)
+    {
+        return adoptRef(*new WebExtensionRegisteredScript(std::forward<Args>(args)...));
+    }
+
+    explicit WebExtensionRegisteredScript(WebExtensionContext& context, String identifier, std::optional<Vector<String>> css, std::optional<Vector<String>> js, std::optional<InjectionTime> injectionTime, std::optional<Vector<String>> excludeMatchPatterns, std::optional<Vector<String>> matchPatterns, std::optional<bool> allFrames, std::optional<bool> persistent, std::optional<bool> forMainWorld)
```

Comment:
had initially tried to pass in the parameters directly but ran into issue. i see that @xeenon did it successfully in his latest PR so will try that approach again